EventTicketingSystem.DataAccess/Extensions/ServiceResolver.cs

```diff
@@ -0,0 +1,57 @@
+﻿using EventTicketingSystem.DataAccess.Interfaces;
+using EventTicketingSystem.DataAccess.Models.Context;
+using EventTicketingSystem.DataAccess.Models.Settings;
+using EventTicketingSystem.DataAccess.Seed;
+using EventTicketingSystem.DataAccess.Services;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace EventTicketingSystem.DataAccess.Extensions
+{
+    /// <summary>
+    /// Class for registering services
+    /// </summary>
+    public static class ServiceResolver
+    {
+        /// <summary>
+        /// Registers services and settings for Data Access. 
+        /// </summary>
+        /// <param name="services">Collection of registered services</param>
+        /// <param name="configuration">Application configuration properties with configuration section for <see cref="EmployeeIdentificationSettings"/></param>
+        public static void AddDataAccess(this IServiceCollection services, IConfiguration configuration)
+        {
+            services.RegisterContext(configuration);
+            services.RegisterRepositories();
+
+            services.AddScoped<ISeedService, SeedService>();
+            services.AddAutoMapper(AppDomain.CurrentDomain.GetAssemblies());
+        }
+
+        private static void RegisterContext(this IServiceCollection services, IConfiguration configuration)
+        {
+            ArgumentNullException.ThrowIfNull(configuration);
```

Comment:
enabling .NET 6 feature nullable reference classes will have the same impact + check is missed in AddDataAccess method

Reply:
Added check in AddDataAccess